SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialDebugger.cpp

```diff
@@ -627,37 +634,135 @@ void ASpatialDebugger::DrawDebug(UCanvas* Canvas, APlayerController* /* Controll
+												MousePosition.Y - (CrosshairTexture->GetSurfaceHeight() * 0.5f));
+
+				// Draw the crosshair at the mouse position.
+				FCanvasTileItem TileItem(CrossHairDrawPosition, CrosshairTexture->Resource, FLinearColor::White);
+				TileItem.BlendMode = SE_BLEND_Translucent;
+				Canvas->DrawItem(TileItem);
+				bCursorChanged = true;
+			}
+
+			if (bCursorChanged)
+			{
+				// Hide the mouse cursor as we will draw our own custom crosshair
+				LocalPlayerController->bShowMouseCursor = false;
+				// Sets back the focus to the game viewport - need to hide mouse cursor instantly
+				FSlateApplication::Get().SetAllUserFocusToGameViewport();
+			}
+
+			// Use the mouse position as the point for sending the raycast into the world
+			if (const APlayerCameraManager* CameraManager = Cast<APlayerCameraManager>(LocalPlayerController->PlayerCameraManager))
+			{
+				FVector WorldLocation;
+				FVector WorldRotation;
+				LocalPlayerController->DeprojectScreenPositionToWorld(MousePosition.X, MousePosition.Y, WorldLocation,
+																	  WorldRotation); // Mouse cursor position
+				FVector StartTrace = WorldLocation;
+				FVector EndTrace = StartTrace + WorldRotation * MaxRange;
+
+				FCollisionObjectQueryParams CollisionObjectParams;
+				CollisionObjectParams.AddObjectTypesToQuery(ECollisionChannel::ECC_WorldStatic);
+				CollisionObjectParams.AddObjectTypesToQuery(ECollisionChannel::ECC_WorldDynamic);
+				CollisionObjectParams.AddObjectTypesToQuery(ECollisionChannel::ECC_Pawn);
+				CollisionObjectParams.AddObjectTypesToQuery(ECollisionChannel::ECC_Destructible);
+				CollisionObjectParams.AddObjectTypesToQuery(ECollisionChannel::ECC_Vehicle);
+				CollisionObjectParams.AddObjectTypesToQuery(ECollisionChannel::ECC_PhysicsBody);
+
+				TArray<FHitResult> HitResults;
+				bool bHit = GetWorld()->LineTraceMultiByObjectType(HitResults, StartTrace, EndTrace, CollisionObjectParams);
+				if (bHit)
 				{
-					continue;
+					// When the raycast hits an actor then the debug information for that actor is displayed above it, whilst the actor
+					// remains under the crosshair.
+					for (const FHitResult& HitResult : HitResults)
+					{
+						const TWeakObjectPtr<AActor> HitActor = HitResult.GetActor();
+
+						if (const Worker_EntityId_Key* HitEntityId = EntityActorMapping.FindKey(HitResult.GetActor()))
+						{
+							FVector PlayerLocation = GetLocalPawnLocation();
+
+							if (HitActor != nullptr)
+							{
+								FVector2D ScreenLocation = ProjectActorToScreen(HitActor, PlayerLocation);
+								if (ScreenLocation.IsZero())
+								{
+									continue;
+								}
+
+								DrawTag(Canvas, ScreenLocation, *HitEntityId, HitActor->GetName(), true /*bCentre*/);
+							}
+						}
+					}
 				}
-
-				DrawTag(Canvas, ScreenLocation, EntityId, Actor->GetName(), true /*bCentre*/);
 			}
 		}
 	}
 }
 
+FVector2D ASpatialDebugger::ProjectActorToScreen(const TWeakObjectPtr<AActor> Actor, const FVector PlayerLocation)
```

Comment:
```suggestion
FVector2D ASpatialDebugger::ProjectActorToScreen(const TWeakObjectPtr<AActor> Actor, const FVector PlayerLocation)
```
- Make the player location const
- I don't see a reason to use an out parameter here, and to me doing so is less readable at the callsites.